pyproject.toml

```diff
@@ -1,147 +1,121 @@
+requires = [
+  "hatchling",
+  "hatch-vcs"
+]
+build-backend = "hatchling.build"
+
 [project]
-  name = "cosmology.compat.camb"
-  version = "0.1.0"
-  description = "Cosmology API standard compatability library for CAMB"
-  readme = "README.rst"
-  requires-python = ">=3.9"
-  license = {file = "LICENSE"}
-  keywords = ["cosmology", "API"]
-  authors = [
-    {name = "Nathaniel Starkman", email = "n.starkman@mail.utoronto.ca"},
-    {name = "Nicolas Tessore", email = "n.tessore@ucl.ac.uk"}
-  ]
-  maintainers = [
-    {name = "Nathaniel Starkman", email = "n.starkman@mail.utoronto.ca"},
-    {name = "Nicolas Tessore", email = "n.tessore@ucl.ac.uk"}
-  ]
-  classifiers = [
-    "Intended Audience :: Science/Research",
-    "Intended Audience :: Developers",
-    "License :: OSI Approved :: BSD License",
-    "Operating System :: OS Independent",
-    "Programming Language :: Python :: 3",
-  ]
-  dependencies = [
-    "cosmology.api",
-    "camb",
-    "numpy>=1.20",
-  ]
+name = "cosmology.compat.camb"
+description = "Cosmology API compatibility layer for CAMB"
+dynamic = [
+  "version"
+]
+readme = "README.rst"
+requires-python = ">=3.10"
+authors = [
+  {name = "Nathaniel Starkman", email = "starkman@mit.edu"},
+  {name = "Nicolas Tessore", email = "n.tessore@ucl.ac.uk"}
+]
+maintainers = [
+  {name = "Nathaniel Starkman", email = "n.starkman@mail.utoronto.ca"},
+  {name = "Nicolas Tessore", email = "n.tessore@ucl.ac.uk"}
+]
+classifiers = [
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3"
+]
+dependencies = [
+  "numpy>=2"
+]
+license.text = "MIT"
 
 [project.optional-dependencies]
-  all = [
-  ]
-  test = [
-    "coverage[toml]",
-    "numpy>=1.19",
-    "pytest",
-    "pytest-astropy",
-  ]
-  docs = [
-    "graphviz",
-    "IPython",
-    "jupyter_client",
-    "nbsphinx",
-    "pydata-sphinx-theme",
-    "sphinx",
-    "sphinx-astropy",
-    "sphinxcontrib.bibtex < 3.0.0",
-    "tomlkit",
-  ]
+doc = [
+  "sphinx",
+  "furo"
+]
+test = [
+  "astropy",
+  "camb",
+  "cosmology.api",
+  "pytest",
+  "pytest-cov"
+]
 
 [project.urls]
-  homepage = "https://cosmology.readthedocs.org/camb"
-  repository = "https://github.com/cosmology-api/cosmology-compat-camb"
-  documentation = "https://cosmology.readthedocs.org/camb"
-
-
-[build-system]
-  requires = [
-    "mypy>=0.991",
-    "setuptools>=45.0",
-    "setuptools_scm[toml]>=6.3.1",
-    "wheel",
-  ]
-
-  build-backend = 'setuptools.build_meta'
-
-
-[tool.setuptools]
-  package-dir = {"" = "src"}
-
-[tool.setuptools.packages.find]
-where = ["src"]
-
-[tool.setuptools_scm]
+Documentation = "https://cosmology.readthedocs.io/projects/camb"
+Repository = "https://github.com/cosmology-api/cosmology.compat.camb"
 
+[tool.coverage.report]
+exclude_also = [
+  "class \\w+\\(Protocol(\\[(\\w+, )*\\w+\\])?\\):", # protocol classes
```

Comment:
Won't this exclude doctests within the Protocol?

Reply:
I'll take it out, it doesn't do anything here since there are no protocols.